JavaScriptSPA/authPopup.js

```diff
@@ -0,0 +1,58 @@
+// Create the main myMSALObj instance
+// configuration parameters are located at authConfig.js
+const myMSALObj = new Msal.UserAgentApplication(msalConfig);
+
+function signIn() {
+  myMSALObj.loginPopup(loginRequest)
+    .then(loginResponse => {
+      console.log(loginResponse);
+      console.log('id_token acquired at: ' + new Date().toString());
+      if (myMSALObj.getAccount()) {
+        showWelcomeMessage(myMSALObj.getAccount());
+      }
+    }).catch((error) => {
+      console.log(error);
+    });
+}
+
+function signOut() {
+  myMSALObj.logout();
+}
+
+function getTokenPopup(request) {
+  return myMSALObj.acquireTokenSilent(request)
+    .catch(error => {
+      console.log("silent token acquisition fails. acquiring token using popup");
+          
+      // fallback to interaction when silent call fails
+        return myMSALObj.acquireTokenPopup(request)
+          .then(tokenResponse => {})
```

Comment:
This will result in nothing getting returned from `getTokenPopup`. Either omit this `.then`, or make sure this chain returns `tokenRespone`.

Reply:
I see. kept it there to keep the pattern:

```javascript
        return myMSALObj.acquireTokenPopup(request)
          .then(tokenResponse => {
            return tokenResponse;
          }).catch(error => {
            console.log(error);
          });
```